server/services/system.ts

```
@@ -582,7 +577,17 @@ private static async getNetworkBandwidth(): Promise<NetworkBandwidth> {
     }
   }
 
+  private static isValidProcessNameOrId(nameOrId: string): boolean {
+    // Allow alphanumeric, hyphens, underscores. Commonly used for names.
+    // PM2 also allows numeric IDs.
+    return /^[a-zA-Z0-9_.-]+$/.test(nameOrId);
+  }
+
   static async restartPM2Process(processName: string): Promise<void> {
+    if (!this.isValidProcessNameOrId(processName)) {
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)